cherab/generomak/plasma/plasma.py

```diff
@@ -215,3 +217,332 @@ def get_edge_plasma(atomic_data=None, parent=None, name="Generomak edge plasma")
+                         th0_fraction=0.8, nh0_decay=10,
+                         timp_core=2.7e3, timp_convexity=2,
+                         timp_concavity=4, nimp_core=5e17,
+                         nimp_convexity=2, nimp_concavity=4,
+                         nimp_decay=8,
+                         vtor_core=1e5, vtor_edge=1e4,
+                         vtor_convexity=2, vtor_concavity=4,
+                         vpol_lcfs=2e4, vpol_decay=0.08):
+    """
+    Returns dictionary of core profile functions and species descriptions
+
+    :param lcfs_values: Dictionary of profile values at the separatrix on outer midplane.
+                        The dictionary has to have the same for, mas the one returned by
+                        the function get_edge_profile_values. The default value is the
+                        dictionary returned by the call get_edge_profile_values for r, z
+                        on last closed flux surface on outer midplane.
+    :param ne_core: core electron density
+    :param ne_convexity: convexity of the electron density profile
+    :param ne_concavity: concavity of the electron density profile
+    :param te_core: core electron temperature 
+    :param te_convexity: convexity of the electron temperature profile
+    :param te_concavity: convexity of the electron temperature profile
+    :param nh_core: core density of H1+
+    :param nh_convexity: convexity of H1+ density profile
+    :param nh_concavity: concavity of H1+ density profile
+    :param th_core: core H1+ temperature
+    :param th_convexity: convexity of H1+ density profile
+    :param th_concavity: convexity of H1+ density profile
+    :param th0_fraction: H0 temperature factor
+    :param nh0_decay: decay rate of H0 density profile
+    :param timp_core: core impurity density
+    :param timp_convexity: convexity of impurity temperature profile
+    :param timp_concavity: concavity of impurity temperature profile
+    :param nimp_core: core impurity density
+    :param nimp_convexity: convexity of impurity density profile
+    :param nimp_concavity: concavity of impurity density profile
+    :param nimp_decay: decay rate of impurity density profile (except bare nuclei)
+    :param vtor_core: toroidal rotation velocity at the edge m/s
+    :param vtor_edge: core toroidal rotation velocity m/s
+    :param vtor_convexity: convexity of the toroidal rotation profile
+    :param vtor_concavity: concavity of the toroidal rotation profile
+    :return: dictionary of Function1D profiles
+    """
+    if lcfs_values is None:
+        # get edge profiles and calculate profile values at midplane outer lcfs
+        equilibrium = load_equilibrium()
+        r = equilibrium.psin_to_r(1)
+        z = 0
+        lcfs_values = get_edge_profile_values(r, z)
+
+    # toroidal rotation profile
+    f1d_vtor = get_double_parabola(vtor_edge, vtor_core, vtor_convexity, vtor_concavity)
+
+    vpol_lcfs = 2e4
+    vpol_decay = 0.08
```

Comment:
`vpol_lcfs` and `vpol_decay` are passed as function arguments already.

Reply:
corrected